src/cdomains/intDomain.ml

```diff
@@ -101,13 +101,66 @@ module Size = struct (* size in bits as int, range as int64 *)
     try int64_of_big_int y with _ -> raise Not_in_int64
 end
 
-module Interval32 : S with type t = (int64 * int64) option = (* signed 32bit ints *)
+module IntPair =
+struct
+  type t = (int64 * int64) [@@deriving to_yojson]
+
+  let min_int, max_int = Size.range Cil.IInt
+
+  let compare (x0, y0) (x1, y1) =
+    match Int64.compare x0 x1 with
+    | 0 -> Int64.compare y0 y1
+    | c -> c
+
+  let leq (x: t option) (y:t option) =
+    match x, y with
+    | None, _ -> true
+    | Some _, None -> false
+    | Some (x1,x2), Some (y1,y2) -> Int64.compare x1 y1 >= 0 && Int64.compare x2 y2 <= 0
+
+  let norm = function None -> None | Some (x,y) ->
+    if Int64.compare x y > 0 then None
+    else if Int64.compare min_int x > 0 || Int64.compare max_int y < 0 then Some (min_int, max_int)
+    else Some (x,y)
+
+  let norm_set x =
```

Comment:
What exactly does this do? Maybe describe it as well?
Also, `norm` and `norm_set` seem to be only used in `Interval32` and `IntervalSet` respectively. So I think it would make sense to put `norm` and `norm_set into the respective modules. This causes a bit of code duplication but is probably the better idea.

Reply:
I just copied as it was in the code before me, nothing more :D I guess, it sets the interval to top if one of its borders lies outside the `[min_int, max_int]`. As for putting `norm` and `norm_set` into the different modules, it looks like a good idea.